src/libsyntax/ast.rs

```diff
@@ -46,7 +45,7 @@ pub struct SyntaxContext(pub u32);
 /// An identifier contains a Name (index into the interner
 /// table) and a SyntaxContext to track renaming and
 /// macro expansion per Flatt et al., "Macros That Work Together"
-#[derive(Clone, Copy, Eq)]
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
```

Comment:
This probably doesn't matter much because the comparisons that changed the meaning could only result in panic before, but still worth checking and maybe fixing later, it's probably not a good idea to compare AST pieces with derived `PartialEq` anyway.